backvendor/vendored.go

```diff
@@ -159,47 +160,66 @@ func updateHashesAfterStrip(hashes *FileHashes, wt *WorkingTree, ref string, pat
 	return anyChanged, nil
 }
 
-func matchFromRefs(strip bool, hashes *FileHashes, wt *WorkingTree, refs []string) (string, error) {
+func matchFromRefs(strip bool, hashes *FileHashes, wt *WorkingTree, refs []string) ([]string, error) {
 	var paths []string
 	if strip {
 		for hash, _ := range hashes.hashes {
 			paths = append(paths, hash)
 		}
 	}
 
-Refs:
+	matchFromRef := func(th *FileHashes, ref string) (bool, error) {
+		if hashes.IsSubsetOf(th) {
+			return true, nil
+		}
+
+		if !strip {
+			return false, nil
+		}
+
+		for _, path := range paths {
+			if _, ok := th.hashes[path]; !ok {
+				// File missing from revision
+				return false, nil
+			}
+		}
+
+		changed, err := updateHashesAfterStrip(th, wt, ref, paths)
```

Comment:
Optional: I know this isn't new code and it's part of a restructuring, but as a user of this function, I wouldn't assume that my passed in `hashes` variable would change, and this method call does that. My preference in the future would be for the passed in `hashes` variable to remain the same and the decision to strip the comments would come from outside of this function where the intent is more apparent.